DietPi-Software | Add MicroK8s install option

silverraindog opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->

Commit:
adding microk8s  install and uninstall

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -1183,6 +1183,13 @@ INDEX_BROWSER=$INDEX_BROWSER"
 		#------------------
+		software_id=142
+
+		aSOFTWARE_NAME[$software_id]='microk8s'
```
```suggestion
		aSOFTWARE_NAME[$software_id]='MicroK8s'
```
We like to use official capitalisation 🙂.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -10385,7 +10392,27 @@ _EOF_
+			# APT deps
+			G_AGI snapd
+		#install
+		G_EXEC snap install microk8s --classic
```
I guess the classic mode is required instead of confinement/jailed mode to allow communication with its containers? Did some reading: https://snapcraft.io/docs/snap-confinement

EDIT: Yes it is, skipping it aborts and prints a related info.

MichaIng replied:
```suggestion
		G_EXEC_OUTPUT=1 G_EXEC snap install microk8s --classic
```
Since this can take a while that users can follow the process. It spams log files, since it prints process bar evolution and such over the same line, while in the log file every single percent creates a new line. But it's worth it for interactive installs, I'd say.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -10385,7 +10392,27 @@ _EOF_
+				[[ -f '/boot/uEnv.txt' ]] && G_CONFIG_INJECT 'docker_optimizations=' 'docker_optimizations=on' /boot/uEnv.txt
+			fi
 
+		fi
```
Check out the updated cgroup handling for K3s above, which applies them on RPi and the new Odroid N2 image (and more of this type to come) as well. I think we should make this a dedicated function, like `Enable_memory_cgroup()` to avoid repeating code.

silverraindog replied:
Thanks, I have taken a look, at the code. I think the function would be a good idea, personally, since in the future, there might be other types of boards that would need to be supported.

I can open an issue for it and add it if you would like ?

MichaIng replied:
Would be okay to do it just in this PR. It is related as the 3rd requirement for the steps satisfies the reason for a dedicated function 😉.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -13434,6 +13461,15 @@ _EOF_
+
+			Banner_Uninstalling
+
+			G_EXEC snap uninstall microk8s 
```
```suggestion
			command -v snap > /dev/null && G_EXEC_OUTPUT=1 G_EXEC snap remove microk8s 
```
It's "remove" not "uninstall", and we shouldn't error out when Snap itself is not installed anymore.

silverraindog replied:
Thought I had fixed it in the IDE but it seems I only fixed it on dietpi-software file on the pi4 and not the IDE

MichaIng committed:
v8.2

- DietPi-Software | MicroK8s: The simplest production-grade upstream K8s. Lightweight and focused, added to DietPi as software option with ID 142. Many thanks to @silverraindog for the implementation: https://github.com/MichaIng/DietPi/pull/5239
